packages/xstate-store/src/index.ts

```diff
@@ -0,0 +1,226 @@
+import type { InteropSubscribable, MachineContext, Observer } from 'xstate';
+import {
+  Recipe,
+  EventPayloadMap,
+  Store,
+  ExtractEventsFromPayloadMap,
+  StoreSnapshot,
+  StorePartialAssigner,
+  StoreCompleteAssigner,
+  StoreAssigner,
+  StorePropertyAssigner
+} from './types';
+
+const symbolObservable: typeof Symbol.observable = (() =>
+  (typeof Symbol === 'function' && Symbol.observable) ||
+  '@@observable')() as any;
+
+export function toObserver<T>(
+  nextHandler?: Observer<T> | ((value: T) => void),
+  errorHandler?: (error: any) => void,
+  completionHandler?: () => void
+): Observer<T> {
+  const isObserver = typeof nextHandler === 'object';
+  const self = isObserver ? nextHandler : undefined;
+
+  return {
+    next: (isObserver ? nextHandler.next : nextHandler)?.bind(self),
+    error: (isObserver ? nextHandler.error : errorHandler)?.bind(self),
+    complete: (isObserver ? nextHandler.complete : completionHandler)?.bind(
+      self
+    )
+  };
+}
+
+function defaultSetter<TContext extends MachineContext>(
+  context: TContext,
+  recipe: Recipe<TContext, TContext>
+): TContext {
+  return recipe(context);
+}
+
+/**
+ * Creates a `Store` that has its own internal state and can receive events.
+ * 
+ * @example
+  ```ts
+  const store = createStore({
+    // initial context
+    { count: 0 },
+    // transitions 
+    {
+      on: {
+        inc: (context, event: { by: number }) => {
+          return {
+            count: context.count + event.by
+          }
+        }
+      }
+    }
+  });
+
+  store.getSnapshot(); // { context: { count: 0 } }
+  store.send({ type: 'inc', by: 5 });
+  store.getSnapshot(); // { context: { count: 5 } }
+  ```
+ */
+export function createStore<
+  TContext extends MachineContext,
+  TEventPayloadMap extends EventPayloadMap
+>(
+  initialContext: TContext,
+  transitions: {
+    [K in keyof TEventPayloadMap & string]:
+      | StoreAssigner<NoInfer<TContext>, { type: K } & TEventPayloadMap[K]>
+      | StorePropertyAssigner<
+          NoInfer<TContext>,
+          { type: K } & TEventPayloadMap[K]
+        >;
+  },
+  updater?: (
+    context: NoInfer<TContext>,
+    recipe: (context: NoInfer<TContext>) => NoInfer<TContext>
+  ) => NoInfer<TContext>
+): Store<TContext, ExtractEventsFromPayloadMap<TEventPayloadMap>> {
+  type StoreEvent = ExtractEventsFromPayloadMap<TEventPayloadMap>;
+  const setter = updater ?? defaultSetter;
```

Comment:
nit: if im not mistaken u don't need this. Whenever you call `setter` you are actually calling `defaultSetter`